lib/ansible/modules/cloud/amazon/ssm_parameter_store.py

```diff
@@ -0,0 +1,248 @@
+                    'metadata_version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: ssm_parameter_store
+short_description: Manage key-value pairs in aws parameter store.
+description:
+  - Manage key-value pairs in aws parameter store.
+version_added: "2.4"
+options:
+  name:
+    description:
+      - parameter key name.
+    required: true
+  description:
+    description:
+      - parameter key desciption.
+    required: false
+  value:
+    description:
+      - Parameter value.
+    required: false
+  state:
+    description:
+      - Creates or modifies an existing parameter
+      - Deletes a parameter
+    required: false
+    choices: ['present', 'absent', 'show']
+    default: present
+  string_type:
+    description:
+      - Parameter String type
+    required: false
+    choices: ['String', 'StringList', 'SecureString']
+    default: String
+  decryption:
+    description:
+      - Work with SecureString type to get plain text secrets
+      - Boolean
+    required: false
+    default: True
+  key_id:
+    description:
+      - aws KMS key to decrypt the secrets.
+    required: false
+    default: aws/ssm (this key is automatically generated at the first parameter created).
+  overwrite:
+    description:
+      - Overwrite the value when create or update parameter
+      - Boolean
+    required: false
+    default: True
+author: Bill Wang (ozbillwang@gmail.com)
+extends_documentation_fragment: aws
+requirements: [ botocore, boto3 ]
+'''
+
+EXAMPLES = '''
+- name: Create or update key/value pair in aws parameter store
+  ssm_parameter_store:
+    name: "Hello"
+    description: "This is your first key"
+    value: "World"
+
+- name: Delete the key
+  ssm_parameter_store:
+    name: "Hello"
+    state: absent
+
+- name: Create or update secure key/value pair in aws parameter store
+  ssm_parameter_store:
+    name: "Hello"
+    description: "This is your first key"
+    string_type: "SecureString"
+    value: "World"
+
+- name: Retrieving plain-text secret
+  ssm_parameter_store:
+    name: "Hello"
+    state: show
+  register: result
+
+- name: Retrieving plain-text secret with custom kms key
+  ssm_parameter_store:
+    name: "Hello"
+    key_id: "aws/ssm"
+    state: show
+  register: result
+
+- name: Retrieving secret without decrypted
+  ssm_parameter_store:
+    name: "Hello"
+    decryption: False
+    state: show
+  register: result
+'''
+
+RETURN = '''
+put_parameter:
+    description: Add one or more paramaters to the system.
+    returned: success
+    type: dictionary
+get_parameter:
+    description: Get details of a parameter.
+    returned: success
+    type: dictionary
+    contains:
+        name:
+            description: The name of the parameter.
+            returned: success
+            type: string
+            sample: "Hello"
+        type:
+            description: The type of parameter. Valid values include the following: String, String list, Secure string..
```

Comment:
YAML check doesn't like this, I would propose to change it to:

`description: The type of parameter. Valid values include [ String, StringList, SecureString ]`